pydantic/_internal/_generate_schema.py

```diff
@@ -2216,11 +2277,34 @@ def _apply_single_annotation_json_schema(
             )
         return schema
 
+    def _get_unsupported_field_info_attributes(self, field_info: FieldInfo) -> list[tuple[str, Any]]:
+        """Get the list of unsupported `FieldInfo` attributes when not directly used in `Annotated` for field annotations."""
+        unused_metadata: list[tuple[str, Any]] = []
+        for unused_metadata_name in UNSUPPORTED_STANDALONE_FIELDINFO_ATTRIBUTES:
+            if (
+                unused_metadata_name in field_info._attributes_set
```

Comment:
If we make `UNSUPPORTED_STANDALONE_FIELDINFO_ATTRIBUTES` a set as well, we could potentially clean this up a bit by using `unsupported_attributes = UNSUPPORTED_STANDALONE_FIELDINFO_ATTRIBUTES & field_info._attributes_set` and then iterating over those

Edit: Likely not relevant because we added the default values to `UNSUPPORTED_STANDALONE_FIELDINFO_ATTRIBUTES` in a later commit